manifests/init.pp

```diff
@@ -1,49 +1,120 @@
+#         'enabled'    => true,
+#         'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#   }
 #
-# [*init_style*]
-#   What style of init system your system uses.
+# @example Disable install and service components
+#   class { '::nomad':
+#     install_method => 'none',
+#     init_style     => false,
+#     manage_service => false,
+#     config_hash   => {
+#       'region'     => 'us-west',
+#       'datacenter' => 'ptk',
+#       'log_level'  => 'INFO',
+#       'bind_addr'  => '0.0.0.0',
+#       'data_dir'   => '/opt/nomad',
+#       'client'     => {
+#       'enabled'    => true,
+#       'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#   }
 #
-# [*purge_config_dir*]
-#   Purge config files no longer generated by Puppet
+# @param arch
+#     cpu architecture
+# @param manage_user
+#     manage the user that will run nomad
+# @param user
+#     username
+# @param manage_group
+#     manage the group that will run nomad
+# @param extra_groups
+#     additional groups to add the nomad user
+# @param purge_config_dir
+#     Purge config files no longer generated by Puppet
+# @param group
+#     groupname
+# @param join_wan
+#     join nomad cluster over the WAN
+# @param bin_dir
+#     location of the nomad binary
+# @param version
+#     Specify version of nomad binary to download.
+# @param install_method
+#     install via system package, download and extract from a url.
+# @param os
+#     operation system to install for
+# @param download_url
+#     download url to download from
+# @param download_url_base
+#     download hostname to down from
+# @param download_extension
+#     archive type to download
+# @param package_name
+#     Only valid when the install_method == package.
+# @param package_ensure
+#     Only valid when the install_method == package.
+# @param config_dir
+#     location of the nomad configuration
+# @param extra_options
+#     Extra arguments to be passed to the nomad agent
+# @param config_hash
+#     Use this to populate the JSON config file for nomad.
+# @param config_defaults
+#     default set of config settings
+# @param config_mode
+#     Use this to set the JSON config file mode for nomad.
+# @param manage_service
+#     manage the nomad service
+# @param pretty_config
+#     Generates a human readable JSON config file.
+# @param pretty_config_indent
+#     Toggle indentation for human readable JSON file.
+# @param service_enable
+#     enable the nomad service
+# @param service_ensure
+#     ensure the state of the nomad service
+# @param restart_on_change
+#     Determines whether to restart nomad agent on $config_hash changes. This will not affect reloads when service, check or watch configs change.
+# @param init_style
```

Comment:
Possible future improvement: should we make it an `Enum['systemd', 'launchd']` instead of `String` to indicate which values are supported? There's already a `fail()` anyway later on in the code.

Reply:
:+1:

Reply:
at the moment my plan was to not want to "fix" to many problems but to get the first vox release ASAP